File: ros2_ws/src/my_python_pkg_example/my_python_pkg_example/simple_node.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node

class My_Node(Node):
    def __init__(self):
        super().__init__("simple_node")
        self.counter_ = 0
        self.create_timer(0.5,self.timer_callback)
        
        
    def timer_callback(self):
        self.get_logger().info("helo world " + str(self.counter_))
        self.counter_ +=1
        
        
def main(args = None):
    rclpy.init(args=args)
    my_node = My_Node()
    rclpy.spin(node=my_node)
    rclpy.shutdown()
    
if __name__ == "__main__":
    main()

File: ros2_ws/src/my_cpp_pkg_example/src/simple_node.cpp
#include <rclcpp/rclcpp.hpp>

class MyNode:public rclcpp::Node
{
public:
    MyNode(): Node("simple_node"), counter_(0)
    {
        RCLCPP_INFO(this->get_logger(),"Node Started");

        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                        std::bind(&MyNode::timer_callback,this));
        
    }

private:
    void timer_callback()
    {
        RCLCPP_INFO(this->get_logger(),"Hello world %d",counter_);
        counter_ ++;
    }
    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};




int main(int argc, char **argv){
    rclcpp::init(argc,argv);
    // auto node = std::make_shared<rclcpp::Node>("simple_node");
    // RCLCPP_INFO(node->get_logger(),"Hello world");

    auto node = std::make_shared<MyNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}